hack/tpm2/main.go

```diff
@@ -0,0 +1,809 @@
+	if err != nil {
+		return nil, err
+	}
+
+	srk, err := tpm2.Unmarshal[tpm2.TPM2BName](luks.TPM2SRK)
+	if err != nil {
+		return nil, err
+	}
+
+	// we need to create a primary since we don't persist the SRK
+	primary := tpm2.CreatePrimary{
+		PrimaryHandle: tpm2.TPMRHOwner,
+		InPublic:      tpm2.New2B(tpm2.ECCSRKTemplate),
+	}
+
+	createPrimaryResponse, err := primary.Execute(t)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		flush := tpm2.FlushContext{
+			FlushHandle: createPrimaryResponse.ObjectHandle,
+		}
+
+		_, flushErr := flush.Execute(t)
+		if flushErr != nil {
+			err = flushErr
+		}
+	}()
+
+	outPub, err := createPrimaryResponse.OutPublic.Contents()
+	if err != nil {
+		return nil, err
+	}
+
+	if !bytes.Equal(createPrimaryResponse.Name.Buffer, srk.Buffer) {
+		// this means the srk name does not match, possibly due to a different TPM or tpm was reset
+		// could also mean the disk was used on a different machine
+		return nil, fmt.Errorf("srk name does not match")
+	}
+
+	load := tpm2.Load{
+		ParentHandle: tpm2.NamedHandle{
+			Handle: createPrimaryResponse.ObjectHandle,
+			Name:   createPrimaryResponse.Name,
+		},
+		InPrivate: *tpmPriv,
+		InPublic:  *tpmPub,
+	}
+
+	loadResponse, err := load.Execute(t)
+	if err != nil {
+		return nil, err
+	}
+
+	policySess, policyCloseFunc, err := tpm2.PolicySession(
+		t,
+		tpm2.TPMAlgSHA256,
+		16,
+		tpm2.Salted(createPrimaryResponse.ObjectHandle, *outPub),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create policy session: %v", err)
+	}
+
+	defer policyCloseFunc()
+
+	pubKey, err := parsePCRSigningPubKey()
+	if err != nil {
+		return nil, err
+	}
+
+	loadExternal := tpm2.LoadExternal{
+		Hierarchy: tpm2.TPMRHOwner,
+		InPublic: tpm2.New2B(tpm2.TPMTPublic{
+			Type:    tpm2.TPMAlgRSA,
+			NameAlg: tpm2.TPMAlgSHA256,
+			ObjectAttributes: tpm2.TPMAObject{
+				Decrypt:      true,
+				SignEncrypt:  true,
+				UserWithAuth: true,
+			},
+			Parameters: tpm2.NewTPMUPublicParms(tpm2.TPMAlgRSA, &tpm2.TPMSRSAParms{
+				Symmetric: tpm2.TPMTSymDefObject{
+					Algorithm: tpm2.TPMAlgNull,
+					Mode:      tpm2.NewTPMUSymMode(tpm2.TPMAlgRSA, tpm2.TPMAlgNull),
+				},
+				Scheme: tpm2.TPMTRSAScheme{
+					Scheme: tpm2.TPMAlgNull,
+					Details: tpm2.NewTPMUAsymScheme(tpm2.TPMAlgRSA, &tpm2.TPMSSigSchemeRSASSA{
+						HashAlg: tpm2.TPMAlgNull,
+					}),
+				},
+				KeyBits:  tpm2.TPMKeyBits(pubKey.N.BitLen()),
+				Exponent: uint32(pubKey.E),
+			}),
+			Unique: tpm2.NewTPMUPublicID(tpm2.TPMAlgRSA, &tpm2.TPM2BPublicKeyRSA{
+				Buffer: pubKey.N.Bytes(),
+			}),
+		}),
+	}
+
+	loadExternalResponse, err := loadExternal.Execute(t)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load external key: %v", err)
+	}
+
+	defer func() {
+		flush := tpm2.FlushContext{
+			FlushHandle: loadExternalResponse.ObjectHandle,
+		}
+
+		_, flushErr := flush.Execute(t)
+		if flushErr != nil {
+			err = flushErr
+		}
+	}()
+
+	pcrSelector, err := createPCRSelection([]int{11})
```

Comment:
`createPCRSelection` was copied from the `go-tpm` tests, not sure if there's another way to create the byte representation form